src/app/api/common/delegates/getDelegates.ts

```diff
@@ -101,22 +102,46 @@ async function getDelegates({
       : "";
 
   let delegateUniverseCTE: string;
-
   const tokenAddress = contracts.token.address;
 
-  delegateUniverseCTE = `with del_statements as (select address from agora.delegate_statements where dao_slug='${slug}'),
-                              del_with_del as (select * from ${namespace + ".delegates"} d where contract = '${tokenAddress}'),
-                              del_card_universe as (select COALESCE(d.delegate, ds.address) as delegate, 
-                                      coalesce(d.num_of_delegators, 0) as num_of_delegators, 
-                                      coalesce(d.direct_vp, 0) as direct_vp, 
-                                      coalesce(d.advanced_vp, 0) as advanced_vp,
-                                      coalesce(d.voting_power, 0) as voting_power
-                                      from del_with_del d full join del_statements ds on d.delegate = ds.address)`;
+  delegateUniverseCTE = `
+    with del_statements as (
+      select address
+      from agora.delegate_statements
+      where dao_slug='${slug}'
+    ),
+    filtered_delegates as (
+      select d.*, vs.participation_rate
+      from ${namespace}.delegates d
+      left join ${namespace}.voter_stats vs
+        on vs.voter = d.delegate
+        and vs.contract = '${contracts.governor.address}'
+      where d.contract = '${tokenAddress}'
+      ${
+        filters?.delegatee
+          ? `
+        AND d.delegate IN (
+          SELECT delegatee
+          FROM ${namespace}.delegatees
+          WHERE delegator = '${filters.delegatee.toLowerCase()}'
+        )
+      `
+          : ""
+      }
+    ),
```

Comment:
I think this is severely slowing down the app for OP (they have 200k+ delegates) 
I was testing this in ENS, and it worked okay (30k records) but I'm noticing the query taking minutes to resolve in OP. Not acceptable imo, I need someone who is good at sql to help take a look at this though.